--- a/hardware/plugins/PluginManager.cpp
+++ b/hardware/plugins/PluginManager.cpp
@@ -385,7 +385,8 @@
 	void CPluginSystem::DeviceModified(uint64_t ID)
 	{
 		std::vector<std::vector<std::string> > result;
-		result = m_sql.safe_query("SELECT HardwareID,Unit FROM DeviceStatus WHERE (ID == %" PRIu64 ")", ID);
+#define	__STDC_FORMAT_MACROS
+		result = m_sql.safe_query("SELECT HardwareID,Unit FROM DeviceStatus WHERE (ID == %" PRIu64 ")", ID);
 		if (!result.empty())
 		{
 			std::vector<std::string> sd = result[0];

